build-logic/src/main/kotlin/ShadowJarExecutableTask.kt

```diff
@@ -0,0 +1,78 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+// Implements https://github.com/brianm/really-executable-jars-maven-plugin maven plugin behaviour.
+// To check details how it works, see https://skife.org/java/unix/2011/06/20/really_executable_jars.html.
+abstract class ShadowJarExecutableTask
+    @Inject
+    constructor(
+        objects: ObjectFactory,
+    ) : DefaultTask() {
+        @InputFile
+        val allJarFile = objects.fileProperty()
+
+        @PathSensitive(PathSensitivity.RELATIVE)
+        @InputFile
+        val windowsBatchScriptSource = objects.fileProperty()
+
+        @Internal
+        val outputDirectory = objects.directoryProperty()
+
+        @OutputFile
+        val selfExecutable = outputDirectory.map { it.asFile.resolve("ktlint") }
+
+        @OutputFile
+        val windowsBatchScript = outputDirectory.map { it.asFile.resolve("ktlint.bat") }
+
+        init {
+            description = "Creates self-executable file, that runs generated shadow jar"
+            group = "Distribution"
+        }
+
+        @TaskAction
+        fun action() {
+            logger.lifecycle("ktlint-cli: Base jar to build self-executable file: ${allJarFile.get()}")
```

Comment:
"Base jar" is confusing to me. Shouldn't this message be identical to the description in the `init` block? Or, alternatively: "Start ShadowJarExecutableTask"?